packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -192,14 +194,27 @@ export class GarbageCollector implements IGarbageCollector {
 			const overrideSessionExpiryTimeoutMs = this.mc.config.getNumber(
 				"Fluid.GarbageCollection.TestOverride.SessionExpiryMs",
 			);
-			const timeoutMs = overrideSessionExpiryTimeoutMs ?? this.configs.sessionExpiryTimeoutMs;
+			let timeoutMs;
+			timeoutMs = overrideSessionExpiryTimeoutMs ?? this.configs.sessionExpiryTimeoutMs;
+
+			if (pendingSessionExpiryTimerStarted) {
+				const timeLapsedSincePendingTimer = Date.now() - pendingSessionExpiryTimerStarted;
+				timeoutMs -= timeLapsedSincePendingTimer;
+			}
+
+			if (timeoutMs <= 0) {
```

Comment:
Something is wrong with the timeout since it does not trigger on my tests even though timeoutMs is negative. Maybe there is a bug in the timer but I can't make the test I added in stashedOps.spec.ts to work correctly without this check.

Reply:
Strange... just checked the Timer impl, it ensures the `setTimeout` call gets a non-negative number (`0` in that case).

Reply:
I think if you add `await delay(0);` after loading in your test, it'll work without this

Reply:
Closing the loop: We decided to keep it so the host's load call will throw.